### PR TITLE
Extract lockfile generation into a new class

### PR DESCRIPTION
**Pull request by segiddins**
### What was the end-user problem that led to this PR?

The problem was that eventually we might want to add more stuff to the lockfile.

### What was your diagnosis of the problem?

My diagnosis was that lockfile generation should probably exist outside of a single method.

### What is your fix for the problem, implemented in this PR?

My fix is to create a `LockfileGenerator` class.

### CODE REVIEW

**Comment by colby-swandale** [comment]
Are there any specs we can extract from `definition_spec` as well? I'm happy for a TODO for the time being.

**Comment by segiddins** [comment]
Nope, nothing that directly tests `to_lock` in `definition_spec`

File: lib/bundler/lockfile_generator.rb [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      add_locked_ruby_version
+      add_bundled_with
+
+      out
+    end
+
+  private
+
+    def add_sources
+      definition.send(:sources).lock_sources.each_with_index do |source, idx|
```
Comment (jules2689) [inline lib/bundler/lockfile_generator.rb]: Is there any need to use `definition.send(:sources)`? Can you access the `sources` directly? If not, can we look at making `sources` not private?

Comment (segiddins) [inline lib/bundler/lockfile_generator.rb]: its not public now and I didn't want to change access control. Eventually, I think all the definition properties that are accessed should be passed into the `LockfileGenerator` but considered it as a second step

**Comment by colby-swandale** [comment]
@bundlerbot r+

**Pull request merged**